cmd/performance-profile-creator/cmd/root.go

```diff
@@ -114,60 +118,100 @@ func getNodeSelectorFromLabelSelector(labelSelector *metav1.LabelSelector) *v1.N
 	return nodeSelector
 
 }
-func getNodes(mustGatherPath string, clientset *ClientSet) ([]*v1.Node, error) {
-	workerNodes := make([]*v1.Node, 0)
 
-	dirs, err := filepath.Glob(mustGatherPath + "/quay-io-openshift-kni-performance-addon-operator-must-gather*/" + "nodes")
+func getMustGatherFullPaths(mustGatherPath string, suffix string) (string, error) {
+	// The regular expression below depends on the must gather image name. It is assumed here
+	// that the image would have "performance-addon-operator-must-gather" substring in the name.
+	paths, err := filepath.Glob(mustGatherPath + "/*performance-addon-operator-must-gather*/" + suffix)
+	// we return only one path here, we could add a validation
+	// 1) If len(paths) == 1 i.e. there is only one possible subdirectory under must gather prefix with the above prefix, we return the path
+	// 2) If len(paths) != 1 we choose one, one option could be to check the timestamp of the directory and choose the most fresh one
+	return paths[0], err
```

Comment:
please note the order on which glob returns is not defined

Reply:
Ack, will make this more sophisticated by checking for the most fresh path

Reply:
I think we should return error if more than one directory is found. We don't know which one the user really wants. Chronological order is just a guess. The error message should be clear: "Expected only one performance-addon-operator-must-gather* directory, please check the must-gather tarball..."